pages/Identity/Registry.md

```diff
@@ -6,21 +6,228 @@ menu: Identity
+
+## EIP 721
+
+DSNP support for the NFT standard [EIP 721](https://eips.ethereum.org/EIPS/eip-721) for handles is not available for [Betanet](/Networks).
+It will be reconsidered for Mainnet.
+
+### NFT Concerns
+
+* EIP 721 ownership standards don't need permissioned levels
+* Dual ownership standard support adds unneeded complexity
+* Identity contract ownership requires receive and transfer support
+
+## Rejected Alternatives
+
+* Why not the Ethereum Name Service ([ENS](https://ens.domains/))?
+  * Overly complex and expensive for the simple use case of handles
+  * Support can be added later through a custom ENS resolver
+  * Didn't provide a numerical id for efficient graph storage
+  * Still subject to homograph attacks
+
+## Contract Interface
+
+```solidity
+/**
+ * @dev DSNP Registry Interface
+ * @dev Suggested data storage implementation:
+ *   uint64 internal currentIdSequenceMarker = 0x1; // Must not start at 0
+ *   mapping(string => [id, address]) internal handleToIdAndAddress;
+ */
+interface IRegistry {
+
+    /**
+     * @dev Log when a resolution address is changed
+     * @param id The DSNP Id 
+     * @param addr The address the DSNP Id is pointing at
+     * @param handle The actual UTF-8 string used for the handle 
+     */
+    event DSNPId(uint64 indexed id, address indexed addr, string indexed handle);
+
+    /**
+     * @dev Register a new DSNP Id
+     * @param addr Address for the new DSNP Id to point at
+     * @param handle The handle for discovery
+     * 
+     * MUST reject if the handle is already in use
+     * MUST be called by someone who is authorized on the contract
+     *      via `IDelegation(addr).isAuthorizedTo(msg.sender, Permission.OWNERSHIP_TRANSFER, block.number)`
+     * MUST emit DSNPId
+     */
+    function register(address addr, string handle) external returns (uint64);
+
+    /**
+     * @dev Register a new DSNP Id by EIP-712 Signature
+     * @param r ECDSA Signature r value
+     * @param s ECDSA Signature s value
+     * @param v EIP-155 calculated Signature v value
+     * @param addr Address for the new DSNP Id to point at
+     * @param handle The handle for discovery
+     * 
+     * MUST reject if the handle is already in use 
+     * MUST be signed by someone who is authorized on the contract
+     *      via `IDelegation(addr).isAuthorizedTo(ecrecovedAddr, Permission.OWNERSHIP_TRANSFER, block.number)`
+     * MUST emit DSNPId
+     */
+    function registerByEIP712Sig(bytes32 r, bytes32 s, uint32 v, address addr, string handle) external returns (uint64);
+
+    /**
+     * @dev Alter a DSNP Id resolution address
+     * @param newAddr Original or new address to resolve to
+     * @param handle The handle to modify
+     * 
+     * MUST be called by someone who is authorized on the contract
+     *      via `IDelegation(oldAddr).isAuthorizedTo(oldAddr, Permission.OWNERSHIP_TRANSFER, block.number)`
+     * TODO: FIX THE ISSUE OF newAddr not being a part of the creation
```

Comment:
what do you mean here - with `newAddr` not being part of creation?

Reply:
oh I see. got it. So then is this still a TODO?

Reply:
@aneyzberg I think it is something we want to consider if we should do anything about it. I added it as an issue: https://github.com/LibertyDSNP/spec/issues/33